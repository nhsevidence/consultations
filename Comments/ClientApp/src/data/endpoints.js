export const BaseUrl  = "/consultations";

export const Endpoints = {
	consultation: "/api/Consultation", // details for a single consultation
	consultations: "/api/Consultations", // list of consultations
	consultationList: "/api/ConsultationList", // list of consultations for the download page
	draftconsultation: "/api/DraftConsultation", // details for a single consultation that has never been published
	documents: "/api/Documents", // documents contained within consultation
	previewdraftdocuments: "/api/PreviewDraftDocuments", // documents contained within consultation
	previewpublisheddocuments: "/api/PreviewpublishedDocuments", // documents contained within consultation
	chapter: "/api/Chapter", // chapter content for a document
	previewchapter: "/api/PreviewChapter", // chapter content for a document in preview mode
	comments: "/api/Comments", // list of comments for a given URI, restricted by current user
	commentsreview: "/api/CommentsForReview", // list of comments for a given URI, restricted by current user
	editcomment: "/api/Comment/{0}", // edits an existing comment. id must be positive int, restricted by current user
	newcomment: "/api/Comment", // creates a new comment. body of message contains comment in json, restricted by current user
	editanswer: "/api/Answer/{0}", // edits an existing answer. id must be positive int, restricted by current user
	newanswer: "/api/Answer", // creates a new answer. body of message contains answer in json, restricted by current user
	review: "/api/Review/{0}", //list of all comments for a consultation (inc Consultation, Document, Chapter and Section levels) given a URI, restricted by current user
	user: "/api/User", // get details of currently logged in user
	submit: "/api/Submit", // submit on the review page.
	submitToOrgLead: "/api/SubmitToOrgLead", // submit to lead on the review page
	exportExternal: "/api/ExportExternal/{0}", //Creates an excel spread sheet for external users of the system that contains only their responses
	logging: "/api/Logging", // sends a message to the server for logging. expects the body to be the message and the loglevel to be in the querystring eg ?logLevel=Error
	questions: "/api/Questions", // get complete page structure for the admin of questions for a given consultation. /api/Questions?consultationId=22
	question: "/api/Question/{0}", // GET PUT for individual questions
	newquestion: "/api/Question", // POST for new questions
	organisation: "/api/Organisation", //GET checks for valid collation code and consultation id in querystring. returns OrganisationCode object
	organisationsession: "/api/Organisation/CreateOrganisationUserSession", //POST. checks for valid collation code and organisation authorisation id in querystring. returns a guid if successful. error otherwise.
	checkorganisationusersession: "/api/Organisation/CheckOrganisationUserSession", //GET. checks for valid session id for a given consultation id. returns a boolean for valid or not.
};
