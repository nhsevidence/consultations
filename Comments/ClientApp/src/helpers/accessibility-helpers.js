/**
 * Pull focus to an element, principally for accessibility
 * @param {string} selection - The ***ID ONLY*** If it's not an interactive element, the target will need an attribute of "tabIndex={-1}"
 */
// todo: needs test
export const pullFocusByQuerySelector = selection => {
	const el = document.querySelector(selection);
	if (el.getAttribute("tabindex") === null) {
		el.setAttribute("tabindex", "-1");
	}
	el.focus();
	console.log(`Pulling focus by querySelector to ${selection}`);
};

export const pullFocusById = selection => {
	selection = selection.replace("#", ""); // make sure there's no # being passed into here
	const el = document.getElementById(selection);
	if (el.getAttribute("tabindex") === null) {
		el.setAttribute("tabindex", "-1");
	}
	el.focus();
	console.log(`Pulling focus by ID to ${selection}`);
};
