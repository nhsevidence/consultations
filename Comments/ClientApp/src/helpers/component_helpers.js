// @flow

import React from "react";
import { HashLink } from "react-router-hash-link";

export const HashLinkTop = (
	label: string,
	destination: string,
	behavior: string = "instant",
	block: string = "start") => {
	return (
		<HashLink
			to={destination}
			scroll={el => el.scrollIntoView({ behavior, block })}>
			{label}
		</HashLink>
	);
};
