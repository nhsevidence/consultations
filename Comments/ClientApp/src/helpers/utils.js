import queryString from "query-string";

/**
 * Create a query string out of an object
 * @param obj the key-value pairs that you'd like to be turned into a querystring
 * @returns {string}
 */
export function objectToQueryString(obj) {
	let str = [];
	for (let p in obj)
		if (obj.hasOwnProperty(p)) {
			str.push(encodeURIComponent(p) + "=" + encodeURIComponent(obj[p]));
		}
	if (str.length) return "?" + str.join("&");
	return "";
}

export function queryStringToObject(qryString) {
	return queryString.parse(qryString);
}

/**
 * Resolve a promise on the next tick - used in testing
 * @see https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#process-nexttick
 * @returns {Promise<any>}
 */
export const nextTick = async () => {
	return new Promise((resolve) => {
		setTimeout(resolve, 0);
	});
};

/**
 * Format a string by replacing {0} with arguments passed in
 * @see https://stackoverflow.com/a/4673436
 * @returns {String}
 */
export function replaceFormat(stringToReplace, args) {
	if (typeof(stringToReplace) === undefined || stringToReplace === null)
		return stringToReplace;

	return stringToReplace.replace(/{(\d+)}/g, function(match, number) {
		return typeof args[number] !== undefined
			? args[number]
			: match
		;
	});
}

/**
 * @param String The string in which you want to check for the presence of "http"
 * Detects link that starts with "http"
 * @returns {Boolean}
 */
export const isHttpLink = link => link.indexOf("http") === 0;
