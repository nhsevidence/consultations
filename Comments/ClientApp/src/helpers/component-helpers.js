// @flow

import React from "react";
import { NavHashLink } from "react-router-hash-link";
import { pullFocusByQuerySelector } from "./accessibility-helpers";

type HashLinkTopType = {
	label: string,
	to: string,
	behavior: string,
	block: string
};

export function HashLinkTop(props: HashLinkTopType) {
	const { label, to, behavior, block } = props;
	return (
		<NavHashLink
			role="menuitem"
			to={to}
			scroll={el => {
				el.scrollIntoView({
					behavior,
					block,
				});
				const element = el.attributes.id.value;
				pullFocusByQuerySelector(`#${element}`);
			}}
		>
			{label}
		</NavHashLink>
	);
}

