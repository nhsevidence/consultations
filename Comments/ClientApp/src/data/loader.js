import {Endpoints, BaseUrl} from "./endpoints";
import axios from "axios";
import https from "https";
import {objectToQueryString, replaceFormat} from "./../helpers/utils";

/**
 * Load data using Axios
 * @see https://github.com/axios/axios
 * @param endpointName - can be either the shortcut of an endpoint as
 * specified in the Endpoints collection, or a URL / file path
 * @param baseUrl - the root url segment that's prepended to the query
 * @param query - the data to be serialised into query string
 * @returns {*|PromiseLike<T>|Promise<T>} - returns a promise with the data
 *
 * Example usage:
 * load("chapter", "/consultations", { consultationId: 1, documentId: 2, chapterSlug: "risk-assessment" });
 * ...would result in...
 * axios("/consultations/api/Chapter?consultationId=1&documentId=2&chapterSlug=risk-assessment").then(res => res.data);
 */

export const generateUrl = (endpointName, baseUrl = BaseUrl, urlParameters = [], query = {}) => {
	var endpoint = Endpoints[endpointName];
	if (!endpoint) return endpointName;

	return baseUrl + replaceFormat(endpoint, urlParameters) + objectToQueryString(query);
};

export const load = (endpoint, baseUrl = BaseUrl, urlParameters = [],  query = {}, method = "GET", data = {}, isJson = false) => {
	return new Promise((resolve, reject) => {
		const url = generateUrl(endpoint, baseUrl, urlParameters, query);
		const headers = isJson ? { "Content-Type" : "application/json"} : {};
		const httpsAgent = (baseUrl.indexOf("https") != -1) ? new https.Agent({ rejectUnauthorized: false }) : {};

		//baseUrl = "https://test.nice.org.uk/consultations";
		//https.globalAgent.options.rejectUnauthorized = false;
		//const httpsAgent = {}; //new https.Agent({ rejectUnauthorized: false });

		console.log(`loader.js: ${url} method: ${method}`);
		axios({url, data, method, headers, httpsAgent})
			.then(response => {
				resolve(response);
			})
			.catch(err => {
				reject(err);
			});
	});
};

export default load;
