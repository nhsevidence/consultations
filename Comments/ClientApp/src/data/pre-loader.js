import { load } from "./loader";
//import stringifyObject from "stringify-object";

// Returns data if it's available or a promise that resolves with the data
// when it's loaded async.
// This assumes the app will be rendered twice on the server: once for
// requests to fire and once when all the data has loaded
const preload = (staticContext, endpoint,  urlParameters = [], query = {}, preloadData = {}, throwOnException = true, method = "GET", headers = {}, content = null, isJson = false) => {
	let data = null;

	// Client - get data from global var
	if (typeof window !== "undefined") {
		if (!window.__PRELOADED__) return null;
		console.log("The end point is: " + endpoint);
		console.log(JSON.stringify(window.__PRELOADED__));
		data = window.__PRELOADED__[endpoint];
		delete window.__PRELOADED__[endpoint]; //this is deleted since the preloaded data should only be used on the initial page render and not persist beyond that
		console.log("Data being returned because type of window is not undefined: " + JSON.stringify(data));
		return data;
	}

	// There should always be a static context on the server but check anyway
	if (!staticContext) {
		console.warn(`Static context was null on the server when executing endpoint: ${endpoint}`);
		return data;
	}

	// Data with that key already preloaded on the server
	if (staticContext.preload.data[endpoint]) {
		console.log("Data being returned because data with that key is already preloaded on the server");
		return staticContext.preload.data[endpoint];
	}
	let cookies = "";
	if (preloadData && preloadData.cookies) {
		cookies = preloadData.cookies;
	}
	// Load fresh data on the server
	const promise = load(endpoint, staticContext.baseUrl, urlParameters, query,  method, content, isJson, cookies, headers)
		.then(response => {
			staticContext.preload.data[endpoint] = response.data;
			console.log("The endpoint from the promise: " + endpoint);
			console.log("The response data: " + JSON.stringify(response.data));
			return response.data;
		})
		.catch(err => {
			console.log("we've hit the catch");
			if (throwOnException){
				// todo: no pages loading on dev / alpha poss to do with the footer erroring...?
				 throw new Error(err);
			}
			else{
			 console.log(err); //this console.log is server-side, so not useful anywhere except locally.
			}
		});

	//staticContext.preload.data[endpoint] = data;

	// Track promises on the static context so that we can wait for all of them to resolve before
	staticContext.preload.loaders.push(promise);
	return null;
};

export default preload;
