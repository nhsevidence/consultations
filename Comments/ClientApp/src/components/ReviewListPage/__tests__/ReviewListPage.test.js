/* global jest */

import React from "react";
import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import toJson from "enzyme-to-json";

import { generateUrl } from "../../../data/loader";
import { nextTick, queryStringToObject } from "../../../helpers/utils";
import ReviewListPageWithRouter, {ReviewListPage} from "../ReviewListPage";
import ConsultationData from "./Consultation";
import DocumentsData from "./Documents";
//import stringifyObject from "stringify-object";

const mock = new MockAdapter(axios);

jest.mock("../../../context/UserContext", () => {
	return {
		UserContext: {
			Consumer: (props) => {
				return props.children({
					isAuthorised: true
				});
			}
		}
	};
});

describe("[ClientApp] ", () => {
	describe("ReviewPage Component", () => {
		const fakeProps = {
			match: {
				url: "/1/review",
				params: {
					consultationId: 1,
				}
			},
			location: {
				pathname: "/1/review",
				search: "?sourceURI=consultations%3A%2F%2F.%2Fconsultation%2F1%2Fdocument%2F1"
			}
		};

		afterEach(() => {
			mock.reset();
		});

		// it("generateDocumentList doesn't filter out documents where convertedDocument is true", async () => {

		// 	const docTypesIn = [
		// 		{ title: "first doc title", sourceURI: "first source uri", convertedDocument : true},
		// 		{ title: "second doc title", sourceURI: "second source uri", convertedDocument : true}];

		// 	const reviewPage = new ReviewListPage(fakeProps);

		// 	const returnValue = reviewPage.generateDocumentList(docTypesIn);

		// 	expect(returnValue.links.length).toEqual(2);
		// });

		// it("generateDocumentList filters out documents where convertedDocument is false", async () => {

		// 	const docTypesIn = [
		// 		{ title: "first doc title", sourceURI: "first source uri", convertedDocument : true},
		// 		{ title: "second doc title", sourceURI: "second source uri", convertedDocument : false}];

		// 	const reviewPage = new ReviewListPage(fakeProps);

		// 	const returnValue = reviewPage.generateDocumentList(docTypesIn);

		// 	expect(returnValue.links.length).toEqual(1);
		// });

		it("queryStringToObject should return an object", async () => {
			const returnValue = queryStringToObject("?search=foo&id=bar");
			expect(returnValue.search).toEqual("foo");
			expect(returnValue.id).toEqual("bar");
		});

		it("should hit the endpoints successfully", async () => {
			const mock = new MockAdapter(axios);

			mock
				.onGet("/consultations/api/Documents?consultationId=1")
				.reply(() => {
					return [200, DocumentsData];
				});

			mock
				.onGet("/consultations/api/Consultation?consultationId=1")
				.reply(() => {
					return [200, ConsultationData];
				});
		});

		// it.only("should hit the submit endpoint successfully", async done => {

		// 	const mock = new MockAdapter(axios);

		// 	const wrapper = mount(
		// 		<MemoryRouter>
		// 			<ReviewPage {...fakeProps} />
		// 		</MemoryRouter>
		// 	);

		// 	let documentsPromise = new Promise(resolve => {
		// 		mock
		// 			.onGet("/consultations/api/Documents?consultationId=1")
		// 			.reply(() => {
		// 				resolve();
		// 				return [200, DocumentsData];
		// 			});
		// 	});

		// 	let consultationPromise = new Promise(resolve => {
		// 		mock
		// 			.onGet("/consultations/api/Consultation?consultationId=1")
		// 			.reply(() => {
		// 				resolve();
		// 				return [200, ConsultationData];
		// 			});
		// 	});

		// 	mock
		// 		.onPost("/consultations/api/Submit")
		// 		.reply(() => {
		// 			done();

		// 		});

		// 	return Promise.all([
		// 		documentsPromise,
		// 		consultationPromise
		// 	]).then(async () => {
		// 		await nextTick();
		// 		wrapper.update();

		// 		//expect(wrapper.find(ReviewPage).instance().state.isSubmitted).toEqual(false);

		// 		wrapper.find(ReviewPage).instance().submitConsultation();
		// 	});

		// });

		it("should match snapshot with supplied data", () => {
			const mock = new MockAdapter(axios);

			const wrapper = mount(
				<MemoryRouter>
					<ReviewListPage {...fakeProps} />
				</MemoryRouter>
			);

			let documentsPromise = new Promise(resolve => {
				mock
					.onGet("/consultations/api/Documents?consultationId=1")
					.reply(() => {
						resolve();
						return [200, DocumentsData];
					});
			});

			let consultationPromise = new Promise(resolve => {
				mock
					.onGet("/consultations/api/Consultation?consultationId=1&isReview=true")
					.reply(() => {
						resolve();
						return [200, ConsultationData];
					});
			});

			return Promise.all([
				documentsPromise,
				consultationPromise
			]).then(async () => {
				await nextTick();
				wrapper.update();
				expect(
					toJson(wrapper, {
						noKey: true,
						mode: "deep"
					})
				).toMatchSnapshot();
			});
		});

	});
});
