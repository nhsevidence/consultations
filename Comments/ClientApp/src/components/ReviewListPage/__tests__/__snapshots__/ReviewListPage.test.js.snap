// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  ReviewPage Component should match snapshot with supplied data 1`] = `
<div
  className="container"
>
  <div
    className="grid"
  >
    <div
      data-g="12"
    >
      <aside
        className="phase-banner mt--b"
      >
        <span
          className="phase-banner__tag"
          onClick={[Function]}
          role="button"
        >
          <span
            className="tag tag--impact tag--agile"
          >
            beta
          </span>
        </span>
        <span
          className="phase-banner__label"
        >
          NICE Consultations
           is in development which means that some features may not work fully.
           
          <a
            href="https://github.com/nhsevidence/consultations"
            rel="noopener noreferrer"
            target="_blank"
          >
            Report an issue
          </a>
        </span>
      </aside>
      <nav
        aria-label="Breadcrumbs"
      >
        <p
          className="visually-hidden"
        >
          You are here:
        </p>
        <ol
          className="breadcrumbs"
          itemScope={true}
          itemType="http://schema.org/BreadcrumbList"
        >
          <li
            className="breadcrumbs__crumb"
            itemProp="itemListElement"
            itemScope={true}
            itemType="http://schema.org/ListItem"
          >
            <a
              href="/"
            >
              Home
            </a>
            <meta
              content={0}
              itemProp="position"
            />
          </li>
          <li
            className="breadcrumbs__crumb"
            itemProp="itemListElement"
            itemScope={true}
            itemType="http://schema.org/ListItem"
          >
            <a
              href="/guidance/indevelopment/gid-ta10232/consultation/"
            >
              Consultation title
            </a>
            <meta
              content={1}
              itemProp="position"
            />
          </li>
          <li
            className="breadcrumbs__crumb"
            itemProp="itemListElement"
            itemScope={true}
            itemType="http://schema.org/ListItem"
          >
            <a
              href="/consultations/1/1/introduction"
            >
              Consultation documents
            </a>
            <meta
              content={2}
              itemProp="position"
            />
          </li>
        </ol>
      </nav>
      <main
        role="main"
      >
        <div
          className="page-header"
        >
          <undefined />
          <div
            className="clearfix"
          >
            bvc
          </div>
          <div
            className="grid"
            onScroll={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <div
              data-g="12 md:9 md:push:3"
            >
              <div
                className="results-info"
              >
                <h2
                  aria-live="assertive"
                  className="results-info__count"
                  id="results-info-count"
                >
                  Showing 3 questions and 2 comments
                </h2>
              </div>
              <div
                data-qa-sel="comment-list-wrapper"
              >
                <div>
                  <ul
                    className="CommentList list--unstyled"
                  >
                    <li
                      className="CommentBox"
                    >
                      <h1
                        className="CommentBox__title mt--0 mb--d"
                        data-qa-sel="comment-box-title"
                      >
                        Question on: 
                        <span
                          className="text-lowercase"
                        >
                          Consultation
                        </span>
                        <br />
                      </h1>
                      <p>
                        <strong>
                          Which areas will have the biggest impact on practice and be challenging to implement? Please say for whom and why.
                        </strong>
                      </p>
                      <section
                        role="form"
                      >
                        <div
                          className="CommentBox__datestamp mb--d font-weight-bold"
                        >
                          Last Modified:
                           
                          <time
                            dateTime={"2018-08-06T14:41:57.930Z"}
                          >
                            6/8/2018 - 3:41pm
                          </time>
                        </div>
                        <form
                          onSubmit={[Function]}
                        >
                          <div
                            className="form__group form__group--textarea mb--0"
                          >
                            <textarea
                              className="form__input form__input--textarea"
                              data-qa-sel="Comment-text-area"
                              disabled={false}
                              id="Answer11"
                              onChange={[Function]}
                              placeholder="Enter your answer here"
                              value="gg"
                            />
                          </div>
                          <input
                            className="btn ml--0"
                            data-qa-sel="submit-button"
                            disabled={true}
                            type="submit"
                            value="Saved"
                          />
                          <button
                            className="btn mr--0 right"
                            data-qa-sel="delete-comment-button"
                            onClick={[Function]}
                          >
                            <span
                              className="visually-hidden"
                            >
                              Delete this answer
                            </span>
                            <span
                              aria-hidden="true"
                              className="icon icon--trash"
                            />
                          </button>
                        </form>
                      </section>
                    </li>
                    <li
                      className="CommentBox"
                    >
                      <h1
                        className="CommentBox__title mt--0 mb--d"
                        data-qa-sel="comment-box-title"
                      >
                        Question on: 
                        <span
                          className="text-lowercase"
                        >
                          Document
                        </span>
                        <br />
                      </h1>
                      <p>
                        <strong>
                          Would implementation of any of the draft recommendations have significant cost implications?
                        </strong>
                      </p>
                      <section
                        role="form"
                      >
                        <form
                          onSubmit={[Function]}
                        >
                          <div
                            className="form__group form__group--textarea mb--0"
                          >
                            <textarea
                              className="form__input form__input--textarea"
                              data-qa-sel="Comment-text-area"
                              disabled={false}
                              id="Answer-1"
                              onChange={[Function]}
                              placeholder="Enter your answer here"
                            />
                          </div>
                        </form>
                      </section>
                    </li>
                    <li
                      className="CommentBox"
                    >
                      <h1
                        className="CommentBox__title mt--0 mb--d"
                        data-qa-sel="comment-box-title"
                      >
                        Question on: 
                        <span
                          className="text-lowercase"
                        >
                          Document
                        </span>
                        <br />
                      </h1>
                      <p>
                        <strong>
                          Would implementation of any of the draft recommendations have cost implications?
                        </strong>
                      </p>
                      <section
                        role="form"
                      >
                        <form
                          onSubmit={[Function]}
                        >
                          <div
                            className="form__group form__group--textarea mb--0"
                          >
                            <textarea
                              className="form__input form__input--textarea"
                              data-qa-sel="Comment-text-area"
                              disabled={false}
                              id="Answer-1"
                              onChange={[Function]}
                              placeholder="Enter your answer here"
                            />
                          </div>
                        </form>
                      </section>
                    </li>
                  </ul>
                </div>
                <ul
                  className="CommentList list--unstyled"
                >
                  <li
                    className="CommentBox"
                  >
                    <section
                      role="form"
                    >
                      <h1
                        className="CommentBox__title mt--0 mb--d"
                        data-qa-sel="comment-box-title"
                      >
                        Comment on: 
                        <span
                          className="text-lowercase"
                        >
                          Document
                        </span>
                        <br />
                        Unstable angina and NSTEMI
                      </h1>
                      <div
                        className="CommentBox__datestamp mb--d font-weight-bold"
                      >
                        Last Modified:
                         
                        <time
                          dateTime={"2018-08-06T14:47:06.638Z"}
                        >
                          6/8/2018 - 3:47pm
                        </time>
                      </div>
                      <form
                        onSubmit={[Function]}
                      >
                        <div
                          className="form__group form__group--textarea mb--0"
                        >
                          <label
                            className="form__label visually-hidden"
                            htmlFor="Comment5"
                          >
                            Comment on 
                            Document
                            , 
                            Unstable angina and NSTEMI
                          </label>
                          <textarea
                            className="form__input form__input--textarea"
                            data-qa-sel="Comment-text-area"
                            disabled={false}
                            id="Comment5"
                            onChange={[Function]}
                            placeholder="Enter your comment here"
                            value="ffffc"
                          />
                        </div>
                        <input
                          className="btn ml--0"
                          data-qa-sel="submit-button"
                          disabled={true}
                          type="submit"
                          value="Saved"
                        />
                        <button
                          className="btn mr--0 right"
                          data-qa-sel="delete-comment-button"
                          onClick={[Function]}
                        >
                          <span
                            className="visually-hidden"
                          >
                            Delete this comment
                          </span>
                          <span
                            aria-hidden="true"
                            className="icon icon--trash"
                          />
                        </button>
                      </form>
                    </section>
                  </li>
                  <li
                    className="CommentBox"
                  >
                    <section
                      role="form"
                    >
                      <h1
                        className="CommentBox__title mt--0 mb--d"
                        data-qa-sel="comment-box-title"
                      >
                        Comment on: 
                        <span
                          className="text-lowercase"
                        >
                          Chapter
                        </span>
                        <br />
                        Introduction 
                      </h1>
                      <div
                        className="CommentBox__datestamp mb--d font-weight-bold"
                      >
                        Last Modified:
                         
                        <time
                          dateTime={"2018-08-06T13:11:38.764Z"}
                        >
                          6/8/2018 - 2:11pm
                        </time>
                      </div>
                      <form
                        onSubmit={[Function]}
                      >
                        <div
                          className="form__group form__group--textarea mb--0"
                        >
                          <label
                            className="form__label visually-hidden"
                            htmlFor="Comment7"
                          >
                            Comment on 
                            Chapter
                            , 
                            Introduction 
                          </label>
                          <textarea
                            className="form__input form__input--textarea"
                            data-qa-sel="Comment-text-area"
                            disabled={false}
                            id="Comment7"
                            onChange={[Function]}
                            placeholder="Enter your comment here"
                            value="ffff"
                          />
                        </div>
                        <input
                          className="btn ml--0"
                          data-qa-sel="submit-button"
                          disabled={true}
                          type="submit"
                          value="Saved"
                        />
                        <button
                          className="btn mr--0 right"
                          data-qa-sel="delete-comment-button"
                          onClick={[Function]}
                        >
                          <span
                            className="visually-hidden"
                          >
                            Delete this comment
                          </span>
                          <span
                            aria-hidden="true"
                            className="icon icon--trash"
                          />
                        </button>
                      </form>
                    </section>
                  </li>
                </ul>
              </div>
              <div
                className="hero"
              >
                <div
                  className="hero__container"
                >
                  <div
                    className="hero__body "
                  >
                    <div
                      className="hero__copy"
                    >
                      <p
                        className="hero__intro"
                      >
                        You are about to submit your final response to NICE
                      </p>
                      <p>
                        After submission you won't be able to:
                      </p>
                      <ul>
                        <li>
                          edit your response further
                        </li>
                        <li>
                          add any extra information.
                        </li>
                      </ul>
                      <undefined />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              data-g="12 md:3 md:pull:9"
            >
              <div>
                <div />
                <div
                  style={Object {}}
                >
                  <div
                    className="filter-panel"
                  >
                    <button
                      aria-controls="filter-panel-body"
                      aria-expanded={true}
                      className="filter-panel__heading"
                      onClick={[Function]}
                      type="button"
                    >
                      Filter
                    </button>
                    <div
                      aria-hidden={false}
                      className="filter-panel__body"
                      id="filter-panel-body"
                    >
                      <div
                        className="filter-group"
                      >
                        <button
                          aria-controls="group-QuestionsOrComments"
                          aria-expanded={true}
                          className="filter-group__heading gtm-topic-list-filter-group-collapse"
                          onClick={[Function]}
                          type="button"
                        >
                          <div
                            id="group-title-QuestionsOrComments"
                          >
                            Response type
                          </div>
                        </button>
                        <div
                          aria-hidden={false}
                          aria-labelledby="group-title-QuestionsOrComments"
                          className="filter-group__options"
                          id="group-QuestionsOrComments"
                          role="group"
                        >
                          <label
                            className="filter-group__option"
                            htmlFor="filter_QuestionsOrComments_Questions"
                          >
                            <input
                              aria-controls="results-info-count"
                              checked={false}
                              className="gtm-topic-list-filter-select"
                              id="filter_QuestionsOrComments_Questions"
                              name="QuestionsOrComments"
                              onChange={[Function]}
                              title="Response type - Questions"
                              type="checkbox"
                              value="Questions"
                            />
                            Questions
                          </label>
                          <label
                            className="filter-group__option"
                            htmlFor="filter_QuestionsOrComments_Comments"
                          >
                            <input
                              aria-controls="results-info-count"
                              checked={false}
                              className="gtm-topic-list-filter-select"
                              id="filter_QuestionsOrComments_Comments"
                              name="QuestionsOrComments"
                              onChange={[Function]}
                              title="Response type - Comments"
                              type="checkbox"
                              value="Comments"
                            />
                            Comments
                          </label>
                        </div>
                      </div>
                      <div
                        className="filter-group"
                      >
                        <button
                          aria-controls="group-Documents"
                          aria-expanded={true}
                          className="filter-group__heading gtm-topic-list-filter-group-collapse"
                          onClick={[Function]}
                          type="button"
                        >
                          <div
                            id="group-title-Documents"
                          >
                            Document
                          </div>
                        </button>
                        <div
                          aria-hidden={false}
                          aria-labelledby="group-title-Documents"
                          className="filter-group__options"
                          id="group-Documents"
                          role="group"
                        >
                          <label
                            className="filter-group__option"
                            htmlFor="filter_Documents_1"
                          >
                            <input
                              aria-controls="results-info-count"
                              checked={false}
                              className="gtm-topic-list-filter-select"
                              id="filter_Documents_1"
                              name="Documents"
                              onChange={[Function]}
                              title="Document - Unstable angina and NSTEMI"
                              type="checkbox"
                              value="1"
                            />
                            Unstable angina and NSTEMI
                          </label>
                          <label
                            className="filter-group__option"
                            htmlFor="filter_Documents_2"
                          >
                            <input
                              aria-controls="results-info-count"
                              checked={false}
                              className="gtm-topic-list-filter-select"
                              id="filter_Documents_2"
                              name="Documents"
                              onChange={[Function]}
                              title="Document - Glaucoma: diagnosis and management – Full Guideline"
                              type="checkbox"
                              value="2"
                            />
                            Glaucoma: diagnosis and management – Full Guideline
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
